Source/Details/ASBasicImageDownloader.mm

```diff
@@ -46,11 +46,12 @@ @interface ASBasicImageDownloaderContext ()
 @implementation ASBasicImageDownloaderContext
 
 static NSMutableDictionary *currentRequests = nil;
-static ASDN::RecursiveMutex currentRequestsLock;
+// Allocate currentRequestsLock on the heap to prevent destruction at app exit (https://github.com/TextureGroup/Texture/issues/136)
+static ASDN::StaticMutex& currentRequestsLock = *new ASDN::StaticMutex;
```

Comment:
I double-checked the code here and I'm 99% sure that this mutex doesn't need to be recursive.

Reply:
^ @garrettmoon

Reply:
Cool :)